cpp/include/cugraph/graph.hpp

```diff
@@ -156,10 +156,12 @@ class graph_t<vertex_t, edge_t, weight_t, store_transposed, multi_gpu, std::enab
 
   bool is_weighted() const { return edge_partition_weights_.has_value(); }
 
-  graph_view_t<vertex_t, edge_t, weight_t, store_transposed, multi_gpu> view() const
+  graph_view_t<vertex_t, edge_t, weight_t, store_transposed, multi_gpu> view(
+    std::optional<std::vector<graph_mask_t<bool> const*>> mask = std::nullopt) const
```

Comment:
What's the purpose of passing `mask` as an input parameter?

Who owns the memory block for masks?

How users will update the mask values?

Based on the answer, `enable_vertex_mask = false` & `enable_edge_mask` might be sufficient?